src/util/key.rs

```diff
@@ -78,6 +78,23 @@ pub struct PublicKey {
 }
 
 impl PublicKey {
+    /// Constructs compressed ECDSA PublicKey from the provided generic Secp256k1 public key
+    pub fn compressed(pk: secp256k1::PublicKey) -> PublicKey {
```

Comment:
I would prefer just `new` for a compressed key (same for private)

Reply:
Will agree. May be also `legacy` for `uncompressed`?

Reply:
For uncompressed I personally haven't a preference both uncompressed or legacy are ok

Reply:
I'd prefer `new` and `new_uncompressed`